api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -387,18 +406,181 @@ public boolean supportsPropertyName(String gpName) {
 	}
 
 	@Override
-	public String importMpiPatient(String personId) {
-		if (registrationCoreProperties.isMpiEnabled()) {
-			MpiProvider mpiProvider = registrationCoreProperties.getMpiProvider();
-			Patient importedPatient = mpiProvider.fetchMpiPatient(personId);
-			Patient patient = patientService.savePatient(importedPatient);
-			return patient.getUuid();
+	public MpiPatient findMpiPatient(String identifier, String identifierTypeUuid) {
+		if (!registrationCoreProperties.isMpiEnabled()) {
+			throw new MpiException("Should not perform 'findMpiPatient' when MPI is disabled");
+		}
+		MpiProvider mpiProvider = registrationCoreProperties.getMpiProvider();
+		MpiPatient mpiPatient = null;
+		try {
+			mpiPatient = mpiProvider.fetchMpiPatient(identifier, identifierTypeUuid);
+		} catch (RuntimeException e) {
+			servePdqExceptionAndThrowAgain(e,
+					prepareParameters(identifier, identifierTypeUuid),
+					PdqErrorHandlingService.FIND_MPI_PATIENT_DESTINATION);
+		}
+		return mpiPatient;
+		//TODO throw error if you get more than one patient
+	}
+
+	@Override
+	public Patient importMpiPatient(String personId) {
```

Comment:
This is method is a bit redundant, the method it calls should be the one to call `mpiProperties.getMpiPersonIdentifierTypeUuid()`

Reply:
And then you get rid of it

Reply:
So
`	public Patient importMpiPatient(String personId) {}
`
and
`
	public Patient importMpiPatient(String patientIdentifier, String patientIdentifierTypeUuid) {}
`
Perform two different functions right? in the first case where you only provide it with the string personId then it assumes that that is the MPI global identifier. in the second case you dont necessarily have to import the mpi patient based on the MPI global identifier, you could do it on any other identifier that may be in the MPI, so its more flexible.

So I think both methods are useful. what do you think?